onsp_co2/views/carbon_factor_source.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <data>
+
+        <record id="carbon_factor_source_tree" model="ir.ui.view">
+            <field name="name">carbon.factor.source.tree</field>
+            <field name="model">carbon.factor.source</field>
+            <field name="arch" type="xml">
+                <tree>
+                    <field name="source_code"/>
+                    <field name="source_name"/>
+                </tree>
+            </field>
+        </record>
+
+        <record id="carbon_factor_source_form" model="ir.ui.view">
+            <field name="name">carbon.factor.source.form</field>
+            <field name="model">carbon.factor.source</field>
+            <field name="arch" type="xml">
+                <form>
+                    <sheet>
+                    <h1><field name="source_name" /></h1>
+                    <group>
+                        <field name="source_code"/>
+                        <!-- <field name="source_name"/> -->
+                        <field name="source_author"/>
+                        <field name="source_url"/>
```

Comment:
Don't forget to rename fields if you do it in python.

Also for the url, you can put a widget to feel like a pro:
```xml
<field name="url" widget="url" />
```

Reply:
Fix in the new commit